numpy/lib/recfunctions.py

```diff
@@ -899,8 +936,8 @@ def join_by(key, r1, r2, jointype='inner', r1postfix='1', r2postfix='2',
         raise ValueError(msg)
 
     # Make temporary arrays of just the keys
-    r1k = drop_fields(r1, [n for n in r1names if n not in key])
-    r2k = drop_fields(r2, [n for n in r2names if n not in key])
```

Comment:
is `drop_fields` still used elsewhere - and if so, should it be replaced with something similar?